### PR TITLE
Add FocusState to Login pages for keyboard toolbar

### PR DESCRIPTION
**Pull request by lsherwin10**

### CODE REVIEW

File: DiveMeets/DiveMeets/DiveMeets.entitlements [hunk 1]
```diff
@@ -1,8 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
-<dict>
-	<key>com.apple.developer.default-data-protection</key>
-	<string>NSFileProtectionComplete</string>
-</dict>
+<dict/>
```
Comment (spencerdearman) [inline DiveMeets/DiveMeets/DiveMeets.entitlements:4]: What is this doing in removing the <key>com.apple.developer.default-data-protection</key>, is this just where we removed that extra entitlement

File: DiveMeets/DiveMeets/Views/Login/LoginScreen.swift [hunk 2]
```diff
@@ -207,3 +241,33 @@ struct LoginPageSearchView: View {
         }
     }
 }
+
+private extension LoginSearchInputView {
+    var hasReachedStart: Bool {
+        self.focusedField == Field.allCases.first
```
Comment (spencerdearman) [inline DiveMeets/DiveMeets/Views/Login/LoginScreen.swift:249]: Can you just explain what a focus state is, I understand it is a wrapper that helps to track which view is receiving user input, but I am not sure how you're applying it

**Review by spencerdearman — APPROVED** [review]
Code looks good overall, I added two comments asking for explanation

**Comment by lsherwin10** [comment]
Yeah I just removed it so it would run on my phone.

**Pull request merged**